infrastructures/docker/src/main/java/org/eclipse/che/workspace/infrastructure/docker/DockerInternalRuntime.java

```diff
@@ -524,17 +543,19 @@ private void sendStoppedEvent(String machineName) {
     }
 
     static class StartSynchronizer {
```

Comment:
Please add javadocs for this class as it becomes quite complex and it is not obvious what await and complete do.